inspirehep/modules/records/jsonschemas/records/elements/id.json

```diff
@@ -0,0 +1,155 @@
+                },
+                "value": {
+                    "pattern": "\\d{4}-\\d{4}-\\d{4}-\\d{3}[0-9X]",
+                    "type": "string"
+                }
+            },
+            "type": "object"
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "JACOW"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "JACoW-\\d+",
+                    "type": "string"
+                }
+            },
+            "type": "object"
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "KAKEN"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "\\d+",
+                    "type": "string"
+                }
+            },
+            "type": "object"
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "ARXIV"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "\\w+",
+                    "type": "string"
+                }
+            },
+            "type": "object"
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "CERN"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "CCID-\\d+",
+                    "type": "string"
+                }
+            }
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "DESY"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "DESY-\\d+",
+                    "type": "string"
+                }
+            }
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "WIKIPEDIA"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "\\w+",
+                    "type": "string"
+                }
+            }
+        },
+        {
+            "properties": {
+                "type": {
+                    "enum": [
+                        "SLAC"
+                    ],
+                    "type": "string"
+                },
+                "value": {
+                    "pattern": "SLAC-\\d+",
+                    "type": "string"
+                }
+            }
+        }
+    ],
+    "required": [
+        "type",
+        "value"
+    ]
+}
```

Comment:
@kaplun, can you have a look at this taxonomy to see what is wrong and what can be made stricter?